docs/4.1/architecture/auth.md

```diff
@@ -0,0 +1,211 @@
+2. Signature of the certificate authority who issued it.
+3. The expiration date, also known as "time-to-live" or simply TTL.
+4. Additional data, such as the node role, stored as a certificate extension.
+
+## Authentication in Teleport
+
+Teleport uses SSH certificates to authenticate nodes and users within a cluster.
+
+There are two CAs operating inside the Auth Server because nodes and users each need their own certificates. <!--TODO: Why?-->
+
+* The **Node CA** issues certificates which identify a node (i.e. host, server, computer). These certificates are used to add new nodes to a cluster and identify connections coming from the node.
+* The **User CA** issues certificates which identify a User. These certificates are used to authenticate users when they try to connect to a cluster node.
+
+### Issuing Node Certificates
+
+Node Certificates identify a node within a cluster and establish the permissions of the node to access to other Teleport services. The presence of a signed certificate on a node makes it a cluster member.
+
+![Node Joins Cluster](../img/node_join.svg)
+
+1. To join a cluster for the first time, a node must present a "join token" to the auth server. The token can be static (configured via config file) or a dynamic, single-use token generated by [`tctl nodes add`](../cli-docs/#tctl-nodes-add).
+
+    !!! tip "Token TTL":
+        When using dynamic tokens, their default time to live (TTL) is 15 minutes, but it can be reduced (not increased) via [`tctl nodes add --ttl`](../cli-doocs/#tctl-nodes-add) flag.
+
+2. When a new node joins the cluster, the auth server generates a new public/private keypair for the node and signs its certificate. This node certificate contains the node's role(s) (`proxy`, `auth` or `node`) as a certificate extension (opaque signed string).
+
+### Using Node Certificates
+
+![Node Authorization](../img/node_cluster_auth.svg)
+
+All nodes in a cluster can connect to the [Auth Server's API](#auth-api-server) <!--Docs about this--> implemented as an HTTP REST service running over the SSH tunnel. This API connection is authenticated with the node certificate and the encoded role is checked to enforce access control. For example, a client connection using a certificate with only the `node` role won't be able to add and delete users. This client connection would only be authorized to get auth servers registered in the cluster.
+
+### Issuing User Certificates
+
+![Client obtains new certificate](../img/cert_invalid.svg)
+
+The Auth Server uses its User CA to issue user certificates. User certificates are stored on a user's machine in the `~/.tsh/<proxy_host>` directory or by `ssh agent`.
+
+1. To get permission to join a access cluster for the first time a user must provide their username, password, and 2nd-factor token. Users can log in with [`tsh login`](../cli-docs/#tsh-login) or via the Web UI. The Auth Server check these against its identity storage and checks the 2nd factor token.
```

Comment:
`join a cluster for the first time`

Also 2-factor tokens aren't used/supported when logging in via an SSO provider - a note about this might be worthwhile.